sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateTTLSuite.scala

```diff
@@ -0,0 +1,579 @@
+        // validate ttl value is removed in the value state column family
+        AddData(inputStream, InputEvent("k1", "get_ttl_value_from_state", -1, null)),
+        AdvanceManualClock(1 * 1000),
+        CheckNewAnswer(),
+        // validate ttl state still has old ttl value present
+        AddData(inputStream, InputEvent("k1", "get_values_in_ttl_state", -1, null)),
+        AdvanceManualClock(1 * 1000),
+        CheckNewAnswer(OutputEvent("k1", -1, isTTLValue = true, 61000)),
+        // advance clock after older expiration value
+        AdvanceManualClock(30 * 1000),
+        // ensure unexpired value is still present in the state
+        AddData(inputStream, InputEvent("k1", "get", -1, null)),
+        AdvanceManualClock(1 * 1000),
+        CheckNewAnswer(OutputEvent("k1", 1, isTTLValue = false, -1)),
+        // validate that the older expiration value is removed from ttl state
+        AddData(inputStream, InputEvent("k1", "get_values_in_ttl_state", -1, null)),
+        AdvanceManualClock(1 * 1000),
+        CheckNewAnswer()
+      )
+    }
+  }
+
+  test("validate multiple value states - processing time ttl") {
+    withSQLConf(SQLConf.STATE_STORE_PROVIDER_CLASS.key ->
+      classOf[RocksDBStateStoreProvider].getName) {
+      val ttlKey = "k1"
+      val noTtlKey = "k2"
+
+      val inputStream = MemoryStream[InputEvent]
+      val result = inputStream.toDS()
+        .groupByKey(x => x.key)
+        .transformWithState(
+          MultipleValueStatesTTLProcessor(ttlKey, noTtlKey),
+          TimeoutMode.NoTimeouts(),
+          TTLMode.ProcessingTimeTTL())
+
+      val clock = new StreamManualClock
+      testStream(result)(
+        StartStream(Trigger.ProcessingTime("1 second"), triggerClock = clock),
+        AddData(inputStream, InputEvent(ttlKey, "put", 1, Duration.ofMinutes(1))),
+        AddData(inputStream, InputEvent(noTtlKey, "put", 2, Duration.ZERO)),
+        // advance clock to trigger processing
+        AdvanceManualClock(1 * 1000),
+        CheckNewAnswer(),
+        // get both state values, and make sure we get unexpired value
+        AddData(inputStream, InputEvent(ttlKey, "get", -1, null)),
+        AddData(inputStream, InputEvent(noTtlKey, "get", -1, null)),
+        AdvanceManualClock(1 * 1000),
+        CheckNewAnswer(
+          OutputEvent(ttlKey, 1, isTTLValue = false, -1),
+          OutputEvent(noTtlKey, 2, isTTLValue = false, -1)
+        ),
+        // ensure ttl values were added correctly, and noTtlKey has no ttl values
+        AddData(inputStream, InputEvent(ttlKey, "get_ttl_value_from_state", -1, null)),
+        AddData(inputStream, InputEvent(noTtlKey, "get_ttl_value_from_state", -1, null)),
+        AdvanceManualClock(1 * 1000),
+        CheckNewAnswer(OutputEvent(ttlKey, -1, isTTLValue = true, 61000)),
+        AddData(inputStream, InputEvent(ttlKey, "get_values_in_ttl_state", -1, null)),
+        AddData(inputStream, InputEvent(noTtlKey, "get_values_in_ttl_state", -1, null)),
```

Comment:
Can we also add some tests that are stopping and starting the streaming query ?

Reply:
Added Stop/StartStream for existing tests to test this scenario.